common/logisticspipes/gui/GuiGuideBook.java

```diff
@@ -0,0 +1,171 @@
+package logisticspipes.gui;
+
+import java.io.IOException;
+
+import net.minecraft.client.gui.GuiButton;
+import net.minecraft.client.gui.GuiScreen;
+import net.minecraft.client.gui.GuiSlider;
+import net.minecraft.client.gui.inventory.GuiContainer;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.util.ResourceLocation;
+
+import org.lwjgl.Sys;
+
+import logisticspipes.LPConstants;
+import logisticspipes.utils.string.StringUtils;
+
+public class GuiGuideBook extends GuiScreen {
+
+	private static final ResourceLocation GUI_BOOK_TEXTURE = new ResourceLocation(LPConstants.LP_MOD_ID, "textures/gui/guide_book.png");
+
+	private GuiButton nextPageBtn, prevPageBtn, slider;
+	private boolean dragging = false;
+	private int mouseX, mouseY;
+
+	//Book vars
+	private int PAGE = 1;
+	private int PAGE_MAX = 10;
+	private int MAX_LENGTH = 50;
+
+	public GuiGuideBook() {
+		super();
+	}
+
+	/** Draw a background container sized based on x, and y size. This is not 1:1 scale.*/
+	protected void drawGuiBackgroundLayer(int x, int y) {
+		GlStateManager.color(1.0F, 1.0F, 1.0F, 1.0F);
+		mc.renderEngine.bindTexture(GUI_BOOK_TEXTURE);
+		int xRem = x;
+		int yRem = y;
+		int eight = 8;
+		GlStateManager.pushMatrix();
+		GlStateManager.translate(0.0F, 0.0F, -2.0F);
+		this.drawDefaultBackground();
+		GlStateManager.popMatrix();
+		GlStateManager.pushMatrix();
+		GlStateManager.translate(-x*4, -y*4, -1.0F);
+		/**This is the darker background like the inventory has. Check whether this is wanted or not.*/
+		GlStateManager.translate(this.width / 2, this.height / 2, 0.0F);
+		/**TopLeft*/
+		drawTexturedModalRect(0, 0, 0, 0, eight * 4, eight * 4);
+		xRem -= 4;
+		/**TopRight*/
+		drawTexturedModalRect(eight * (x - 4), 0, eight * 4, 0, eight * 4, eight * 4);
+		xRem -= 4;
+		/**BottomLeft*/
+		drawTexturedModalRect(0, eight * (y - 4), 0, eight * 4, eight * 4, eight * 4);
+		yRem -= 4;
+		/**BottomRight*/
+		yRem -= 4;
+		drawTexturedModalRect(eight * (x - 4), eight * (y - 4), eight * 4, eight * 4, eight * 4, eight * 4);
+		/** Top & Bottom/ */
+		for(int i = 0; i < xRem; i++){
+			drawTexturedModalRect(eight * (i + 4), 0, eight * 3, 0, eight * 2, eight * 4);
+			drawTexturedModalRect(eight * (i + 4), eight * (y - 4), eight * 3, eight * 4, eight * 2, eight * 4);
+		}
+		/**Left, Right & Background*/
+		for(int j = 0; j < yRem; j++) {
+			drawTexturedModalRect(0, eight * (j + 4), 0, eight * 3, eight * 4, eight * 2);
+			drawTexturedModalRect(eight * (x - 4), eight * (j + 4), eight * 4, eight * 3, eight * 4, eight * 2);
+			for(int i = 0; i < xRem; i++){
+				drawTexturedModalRect(eight * (i + 4), eight * (j + 4), eight * 3, eight * 3, eight * 2, eight * 2);
+			}
+		}
```

Comment:
All this drawing can be replaced by 9 draw calls only: 4 for the edges, 4 for the sides (stretch the texture horizontally or vertically depending on the side) and 1 for the center (stretch in both directions)

However, that will cause the texture to be stretched, obviously. But that shouldn't matter in this case since it doesn't have any variation (such as noise) in it.

Reply:
Just tell me how to strech it because I didn't use it just because I didn´t know how to

Reply:
`drawScaledCustomSizeModalRect(x, y, u, v, uWidth, vHeight, width, height, 1f / textureWidth, 1f / textureHeight)`

where x,y,width,height are position on screen, and u,v,uWidth,uHeight are texture coordinate related

(and obviously replace textureWidth and textureHeight with the dimensions of the texture file)

Should work.

Reply:
Unless you meant for me to use `GlStateManager.scale()`